agimus_controller_ros/agimus_controller_ros/simple_trajectory_publisher.py

```
@@ -147,21 +115,43 @@ def get_param_from_node(self, node_name: str, param_name: str) -> ParameterValue
         # TODO fix this, temp hac to work from sim
-        if np.linalg.norm(jpos) > 1e-2:
-            joint_idxs = get_joint_idxs(self.moving_joint_names, joint_states_msg)
+        joint_idxs = get_joint_idxs(self.moving_joint_names, joint_states_msg)
+        if self.q0 is None and np.linalg.norm(jpos) > 1e-2:
             self.q0 = get_reduced_configuration(jpos, joint_idxs)
-            self.destroy_subscription(self.state_subscriber)
+            self.trajectory.set_init_configuration(q0=self.q0)
             self.get_logger().warn(f"Received q0 = {[round(el, 2) for el in self.q0]}.")
+        self.current_q = get_reduced_configuration(jpos, joint_idxs)
+        self.current_dq = get_reduced_configuration(
+            np.array(joint_states_msg.velocity), joint_idxs
+        )
 
     def robot_description_callback(self, msg: String) -> None:
         """Create the models of the robot from the urdf string."""
         self.get_logger().warn("Received robot description.")
         self.robot_description_msg = msg
         self.destroy_subscription(self.subscriber_robot_description_)
 
+    def get_trajectory(self, trajectory_name: String) -> TrajectoryBase:
+        """Build chosen trajectory."""
+        if trajectory_name == "sine_wave_configuration_space":
+            return SinusWaveConfigurationSpace(
+                period=4.0,
+                scale_duration=0.8,
+                amp=0.2,
+                ee_frame_name=self.ee_frame_name,
+                w_q=self.get_weights(self.params.w_q, self.croco_nq),
+                w_qdot=self.get_weights(self.params.w_qdot, self.croco_nq),
+                w_qddot=self.get_weights(self.params.w_qddot, self.croco_nq),
+                w_robot_effort=self.get_weights(
+                    self.params.w_robot_effort, self.croco_nq
+                ),
+                w_pose=self.get_weights(self.params.w_pose, 6),
+            )
+        else:
+            self.get_logger().error("Unknown Trajectory.")
```

maybe raise an error here?